azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -65,34 +84,49 @@ public int createImageVersion(ImageVersion imageVersion) {
       imageVersionId = databaseOperator.transaction(insertAndGetSpaceId).intValue();
     } catch (SQLException e) {
       log.error("Unable to create the image version metadata", e);
+      throw  new ImageMgmtDaoException("Exception while creating image version metadata");
     }
     return imageVersionId;
   }
 
   @Override
-  public List<ImageVersion> getImageVersion(RequestContext requestContext) throws ImageMgmtException {
+  public List<ImageVersion> findImageVersions(RequestContext requestContext) throws ImageMgmtException {
     List<ImageVersion> imageVersions = new ArrayList<>();
     try {
       StringBuilder queryBuilder = new StringBuilder(SELECT_IMAGE_VERSION_BASE_QUERY);
+      List<Object> params = new ArrayList<>();
+      // add imageType in the query
       if(requestContext.getParams().containsKey(ImageMgmtConstants.IMAGE_TYPE)) {
         queryBuilder.append(" AND ");
-        queryBuilder.append(" it.type = '"+requestContext.getParams().get(ImageMgmtConstants.IMAGE_TYPE)+"'");
+        queryBuilder.append(" it.name = ?");
+        params.add(requestContext.getParams().get(ImageMgmtConstants.IMAGE_TYPE));
       }
-
+      // add imageVersion in the query if present
```

Comment:
Nit: Add (a capital. Same for subsequent comments.)

Reply:
Done.